app/UI/edit-profile.tsx

```diff
@@ -185,6 +183,37 @@ export class EditProfile extends Component<Props, State> {
         this.newFieldValue.current.value = "";
     };
 
+    uploadImage = (e: h.JSX.TargetedEvent<HTMLInputElement>) => {
+        const { props, state } = this;
+        const selectedFiles = e.currentTarget.files;
+        if (!selectedFiles) {
+            return;
+        }
+        props.emit({
+            type: "UploadImage",
+            file: selectedFiles[0],
+            callback: (uploaded) => {
```

Comment:
I mentioned it before that this `callback` is not a good design. We did not have time to discuss what could be a better design.

We need to do it in the future when working on image/file uploading again.